lms/djangoapps/oauth2_handler/handlers.py

```diff
@@ -192,17 +193,21 @@ def find_courses(self, user, access_type, values=None):
 
     # pylint: disable=missing-docstring
     def _get_courses_with_access_type(self, user, access_type):
-
         # Check the application cache and update if not present. The application
         # cache is useful since there are calls to different endpoints in close
-        # succession, for example the id_token and user_info endpoins.
+        # succession, for example the id_token and user_info endpoints.
 
         key = '-'.join([str(self.__class__), str(user.id), access_type])
         course_ids = cache.get(key)
 
-        if course_ids is None:
-            course_ids = [unicode(course.id) for course in _get_all_courses()
-                          if has_access(user, access_type, course)]
+        if not course_ids:
+            if GlobalStaff().has_user(user):
+                # Global staff have access to all courses, so just return them all.
+                course_ids = [unicode(course.id) for course in _get_all_courses()]
```

Comment:
`get_courses` should only be returning courses. The only other possibility I can think of is that it might have been returning libraries (`ContentLibrary`), but those should only be returned by `get_libraries`. If you really do need to only get courses, then I would recommend using `course.scope_ids.block_type == 'course'`, so that if we convert courses to XBLocks (and rename the class), you won't get screwed up.